opentracing-util/src/main/java/io/opentracing/util/GlobalTracer.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.util;
+
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Forwards all methods to another tracer that can be configured in one of two ways:
+ * <ol>
+ * <li>Explicitly, calling {@link #register(Tracer)} with a configured tracer, or:</li>
+ * <li>Automatically using the Java {@link ServiceLoader} SPI mechanism to load a {@link Tracer} from the classpath.</li>
+ * </ol>
+ * <p>
+ * When the tracer is needed it is lazily looked up using the following rules:
```

Comment:
I've been thinking about GlobalTracer... I definitely like the idea of having some sort of Tracer singleton in the core OT-Java repository / artifact (as it's essential to making something like #111 effective in practice), BUT I find this impl to be surprisingly subtle for an abstraction I'd expect to be simple.

My sense is that the complexity mostly stems from the requirement to lazy-init the GlobalTracer to avoid init-time race conditions. I get that the race conditions are to be avoided... but how about having the application programmer provide a signal about when to initialize GlobalTracer? Along with a convention that such initialization happens very early in the process lifetime, perhaps we could get rid of most/all of the lazy init code (as well as the reentrant lock, etc).

TL;DR I really want to keep it simple with GlobalTracer.